xenstore/xenstore.go

```diff
@@ -222,6 +225,53 @@ func xs_ls(script_name string, args []string) {
 	}
 }
 
+func xs_chmod(script_name string, args []string) {
+	if len(args) < 2 || args[0] == "-h" < 2 {
+		die("Usage: %s key mode [modes...]", script_name)
+	}
+
+	var err error
+	key := args[0]
+	var perms []xenstoreclient.Permission
+
+	for _, m := range args[1:] {
+		if len(m) > 1 {
+			var p xenstoreclient.Permission
+			switch m[0] {
+			case 'n':
+				p.Pe = xenstoreclient.PERM_NONE
+			case 'r':
+				p.Pe = xenstoreclient.PERM_READ
+			case 'w':
+				p.Pe = xenstoreclient.PERM_WRITE
+			case 'b':
+				p.Pe = xenstoreclient.PERM_READWRITE
+			default:
+				err = errors.New("Invalid mode string")
+			}
+			if err == nil {
+				var id uint64
+				id, err = strconv.ParseUint(m[1:], 10, 0)
+				if err == nil {
+					p.Id = uint(id)
+					perms = append(perms, p)
+				}
+			}
+		} else {
+			err = errors.New("Invalid mode string")
```

Comment:
Here the error message `Invalid mode string` is same as the Line 250, when this kind of error appear then we can't determine where is the exact place. Suggest giving a different error message.

Reply:
OK. Set a different err message.